File: String_Functions/generate_all_substrings/gen_substr.py
def func_body(input_str):
    # Generating all substrings
    n = len(input_str)
    for i in range(n):             # Starting index
        for j in range(i, n):      # Ending index
            print(input_str[i:j+1])  # Extract and print substring

if __name__ == "__main__":
    # Main function where the function is called and use case is demonstrated
    input_str = "abcd"  # Example input
    func_body(input_str)

File: String_Functions/generate_all_substrings/gen_substr.java
package generate_all_substrings;

public class gen_substr{
    public static void func_body(String str) {
        int n = str.length();
        // Generating all substrings
        for (int i = 0; i < n; i++) {         
            for (int j = i; j < n; j++) {     
                System.out.println(str.substring(i, j + 1)); 
            }
        }
    }

    public static void main(String[] args) {
        String input = "abcd"; // Example input
        // Call the function with the string as an argument
        func_body(input);
    }
}
